test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -87,10 +147,24 @@ session_simulator::set_prepare_timestamp(uint64_t ts)
     _prepare_ts = ts;
 }
 
-void
-session_simulator::set_read_timestamp(uint64_t ts)
+int
+session_simulator::set_read_timestamp(uint64_t read_ts)
 {
-    _read_ts = ts;
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    WT_SIM_RET(ts_manager->validate_read_timestamp(this, read_ts));
+
+    /*
+     * If the given timestamp is earlier than the oldest timestamp then round the read timestamp to
+     * oldest timestamp.
+     */
+    connection_simulator *conn = &connection_simulator::get_connection();
+    uint64_t oldest_ts = conn->get_oldest_ts();
+    if (_ts_round_read && read_ts < oldest_ts) {
+        _read_ts = oldest_ts;
+    } else
+        _read_ts = read_ts;
```

Comment:
Here, if _ts_round_read is false, it is still setting the read_ts even if it is less than oldest_ts ??

Reply:
validate_read_timestamp() that's called earlier will fail if `_ts_round_read` is false and the `read_ts` is less than the oldest.

Reply:
To avoid confusion, even though I agree that `validate_read_timestamp` checks for that condition, we could do this:
```suggestion
    if (_ts_round_read && read_ts < oldest_ts)
        _read_ts = oldest_ts;
    else if(read_ts >= oldest_ts) {
        _read_ts = read_ts;
```

Reply:
Yeah, that's more clear.